lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1230,19 +1235,23 @@ protected getCachedTokenInternal(scopes : Array<string> , user: User): CacheResu
 
         const scope = scopes.join(" ").toLowerCase();
         const userObject = user ? user : this.getUser();
-        const adalIdToken = this._cacheStorage.getItem(Constants.adalIdToken);
-        //if user is not currently logged in and no login_hint/sid is passed as an extraQueryParamater
-          if (!userObject && Utils.checkSSO(extraQueryParameters) && Utils.isEmpty(adalIdToken) ) {
-              this._logger.info("User login is required");
-              reject(ErrorCodes.userLoginError + Constants.resourceDelimeter + ErrorDescription.userLoginError);
-              return null;
+
+        //if user is unavailable and no SSO hint is provided...
+        if (!userObject && !Utils.hasSsoHint(extraQueryParameters)) {
+          const adalIdToken = this._cacheStorage.getItem(Constants.adalIdToken);
```

Comment:
Sorry I am still a little new to understanding the full flow, so @sameerag  may want to jump in and provide feedback here